builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -189,7 +189,28 @@ func resourceAwsLambdaPermissionRead(d *schema.ResourceData, meta interface{}) e
 		statement, err = findLambdaPolicyStatementById(&policy, d.Id())
 		return resource.RetryableError(err)
 	})
+
 	if err != nil {
+		switch e := err.(type) {
+		case *resource.TimeoutError:
+			switch underlying := e.LastError.(type) {
+			case awserr.Error:
+				if underlying.Code() == "ResourceNotFoundException" {
+					log.Printf("[WARN] No Lambda Permission Policy found: %v", input)
+					d.SetId("")
+					return nil
+				}
+			case *resource.NotFoundError:
+				log.Printf("[WARN] found Lambda Permission Policy for %v but statement '%s' is absent", input, d.Id())
+				d.SetId("")
+				return nil
+			}
+		case (*resource.NotFoundError):
```

Comment:
`NotFoundChecks` gets defaulted to 20, so if the check ran > 20 times in 1 minute, then `NotFoundError` could be returned. If we discard that possibility, we could do:
```go
if err != nil {
  if tErr, ok := err.(*resource.TimeoutError); ok {
    if awsErr, ok := tErr.LastError.(awserr.Error); ok && awsErr.Code() == "ResourceNotFoundException" {
      log.Printf("[WARN] No Lambda Permission Policy found: %v", input)
      d.SetId("")
      return nil
    }

    if nfErr, ok := tErr.LastError.(*resource.NotFoundError); ok {
      log.Printf("[WARN] found Lambda Permission Policy for %v but statement '%s' is absent", input, d.Id())
      d.SetId("")
      return nil
    }
  }

  return err
}
```